tools/clang-reformat.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2019 The STE||AR-Group
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# iterate over all files in the repo and reformat those that must be checked
+
+for file in $(git ls-files | grep -E "\.(cpp|hpp|cu)(\.in)?$"); do
+    # to allow for per-directory clang format files, we cd into the dir first
+    DIR=$(dirname "$file")
+    pushd ${DIR} >/dev/null
+    clang-format -i $(basename -- ${file})
+    popd >/dev/null
+done
```

Comment:
Do we need this script to be checked in?

Reply:
It isn't needed, but it's nice to have these things around occasionally. If you keep them elsewhere, they get lost etc - no?

Reply:
Yeah, we'll need it now (I don't know if your string-merging-script is this one or if it's a separate script, but it can be here).

Reply:
string merging, I will do in another pr once I'm certain all corner cases are handled.